src/Streamly/Internal/Data/Stream/StreamD/Eliminate.hs

```diff
@@ -228,19 +228,6 @@ parse_ (PRD.Parser pstep initial extract) stream@(Stream step state) = do
 -- Specialized Folds
 ------------------------------------------------------------------------------
 
--- | Run a streaming composition, discard the results.
-{-# INLINE_LATE drain #-}
-drain :: Monad m => Stream m a -> m ()
--- drain = foldrM (\_ xs -> xs) (return ())
-drain (Stream step state) = go SPEC state
```

Comment:
Why was this removed/moved?

Reply:
It is required in low level modules like instances of SerialT and other types.